torch/_inductor/codegen/triton.py

```diff
@@ -1669,13 +1745,11 @@ def codegen_block_ptr(
         return block_ptr, advance_block_ptr, other
 
     def codegen_block_ptr_store_line(self, name, indexing, block_ptr, value, other=""):
-        # broadcasting is not implicit for block_ptrs
-        value = (
-            f"tl.broadcast_to({value}, {self.index_to_str(indexing.reshape_suffix)})"
+        # Broadcast and restore singletons.
```

Comment:
I couldn't find an example where this actually restores singletons. Maybe I should change the comment?